source/module_base/test/math_polyint_test.cpp

```diff
@@ -0,0 +1,134 @@
+#include"../math_polyint.h"
+#include"gtest/gtest.h"
+#include"../realarray.h"
+#include<math.h>
+
+#define doublethreshold 1e-9
+
+/************************************************
+*  unit test of class PolyInt
+***********************************************/
+
+/**
+ * This unit test is to verify the accuracy of 
+ * interpolation method on the function sin(x)/x
+ * with a interval of 0.01.
+ * sin(x)/x is one of the solution of spherical bessel
+ * function when l=0.
+ * 
+ * - Tested function:
+ *      - 4 types of Polynomial_Interpolation 
+ *      - Polynomial_Interpolation_xy
+ */
+
+
+class bessell0 : public testing::Test
+{
+    protected:
+    
+    int TableLength = 400;
+    double interval = 0.01;
+    ModuleBase::realArray table3,table4;
+    ModuleBase::realArray y3;
+    double *tablex = new double[TableLength];
+    double *tabley = new double[TableLength];
+
+    double Func(double x) {return sin(x)/x;}
+
+    void SetUp()
+    {
+        table3.create(1,1,TableLength);
+        table4.create(1,1,1,TableLength);
+        y3.create(1,1,TableLength);
+
+        for(int i=1;i<TableLength;++i) 
+        {
+            table3(0,0,i) = Func(i * interval);
+            table4(0,0,0,i) = Func(i * interval); 
+            tablex[i] = i * interval;
+            tabley[i] = Func(i * interval);
+        }
+    }
+
+    void TearDown()
+    {
+        delete [] tablex;
```

Comment:
`TearDown()` should be counterpart of `SetUp()`. However, `tablex` is defined as class member, and this usage may cause memory leak in production code. If you wish to do so, `new double[TableLength]` should be within the constructor of  `class bessell0`, and `delete` in destructor. Here, a better solution is allocating memory in `SetUp()`.